### DISCUSSION TITLE
sync delay

### DISCUSSION THREAD

**Discussion by jwelp70** [SauceLLC/sauce4strava#193] [General]
I just joined Sauce, but have yet to be able to sync activities.  It keeps giving me a delay time, when that time expires it adds another 2 hours or so.  This has been going on for the last 24 hours.  Thanks.

**Comment by mayfield** [comment]
This sounds normal as long as you're still seeing ~200 activities sync per hour and ~700 per day.  Those are the 2 main rate limits that the sync process has to be within to not get throttled by Strava.  Once the first sync is done, which might take a few days if you have a lot of activities, updates are quick and usually happen in the background without delay.  Be sure to backup/export your data when it's done syncing so you can avoid having to do the slow network sync in the future.

**Comment by tobisnet** [comment]
I tried to setup Sauce on a new PC today and my sync is delayed till January 25:

```
18:28:09 DEBUG RateLimiter [streams-min]: period: 65s, usage: 33/30
18:28:09 INFO Rate limited for 41494 minutes
18:28:05 INFO Starting sync job: <Tobi 🐊⚡ Scheller (6500402)>
```

What am I doing wrong?

**Reply by mayfield** [reply]
Hi @tobisnet 

Can you help me triage this? 

If you open devtools on a strava.com page and enter this in the console, I'd like to know the exact output:
```
JSON.stringify(await sauce.storage.get('hist-rate-limiter-streams-day', {sync: true}))
```

**Reply by mayfield** [reply]
Actually there are 3 commands I'd like to see the output for: 

`JSON.stringify(await sauce.storage.get('hist-rate-limiter-streams-min', {sync: true}))`
`JSON.stringify(await sauce.storage.get('hist-rate-limiter-streams-hour', {sync: true}))`
`JSON.stringify(await sauce.storage.get('hist-rate-limiter-streams-day', {sync: true}))`

**Reply by tobisnet** [reply]
```
`JSON.stringify(await sauce.storage.get('hist-rate-limiter-streams-min', {sync: true}))
'{"spec":{"limit":30,"period":65000,"spread":true},"version":2,"zBucketV1":{"start":1735909245074,"zDeltas":"AUoAtf+PEYUR9hCBEYcUgxGrE/gQ3SevdIAR5xHzIpyn6XuCEf0Q/BD6EP0Qmd6hIOIBuwOhoMEx+xDHxdl3hBH5EP8Q+xDtxdhSlAKlBA=="}}'

JSON.stringify(await sauce.storage.get('hist-rate-limiter-streams-hour', {sync: true}))
'{"spec":{"limit":200,"period":4100000},"version":2,"zBucketV1":{"start":1735909245075,"zDeltas":"AUoAtf+OEYYR9hCBEYkUgRGsE/cQ3SeudIER5xHyIp2n6XuCEfwQ/BD6EP4QmN6hIOIBvAOjoMEx+BDIxdl3hRH3EIAR+xDtxdhSlAKlBA=="}}'

JSON.stringify(await sauce.storage.get('hist-rate-limiter-streams-day', {sync: true}))
'{"spec":{"limit":700,"period":90000000},"version":2,"zBucketV1":{"start":1735909245075,"zDeltas":"AUoAtf+PEYYR9RCBEYwU/hCtE/YQ3SeudIER5xHyIp2n6XuCEf0Q+xD7EP0Qmd6hIOIBuwOjoMEx+RDHxdl3hhH2EIIR+hDsxdhSlQKkBA=="}}'`
```

**Comment by mayfield** [comment]
Should hopefully be resolved now with v8.7.11.1.
